### PR TITLE
UpsellNudge: Pass props to Banner explicitly

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* Following up on Kerry's work in #41115, this passes through all props from `UpsellNudge` to be used by `Banner`, and removes the use of `...props`.

Screen Shot 2020-04-15 at 12 52 13 PM

#### Testing instructions

Switch to this PR and follow testing instructions for upsells in these PRs:

#40721
#40626

Make sure you test with a Jetpack site as well as a WordPress.com site, and verify that upsells only show when it makes sense (ie. upsell for the Premium plan should not show when on a Premium or Business plan).

### CODE REVIEW

File: client/blocks/upsell-nudge/index.jsx [hunk 1]
```diff
@@ -60,7 +60,17 @@ export const UpsellNudge = ( {
 		href = addQueryArgs( { feature, plan }, `/plans/${ site.slug }` );
 	}
 
-	return <Banner { ...props } showIcon={ showIcon } className={ classes } href={ href } />;
+	return (
+		<Banner
+			{ ...props }
```
Comment (mmtr) [inline client/blocks/upsell-nudge/index.jsx:65]: I think it'd be safer to be explicit here and pass down only whitelisted props so we don't use `...props` any more in the banner (which I believe was @gwwar's [suggestion](https://github.com/Automattic/wp-calypso/pull/41115/files#r408500126))

Comment (sixhours) [reply client/blocks/upsell-nudge/index.jsx:65]: Ohhh OK, gotcha. I'll do another pass and evaluate the other props and include those explicitly as well. Thanks!

**Comment by gwwar** [comment]
Good spot @keoshi, here's the two with full context:

| Branch | Prod |
| ------------- | ------------- |
| Screen Shot 2020-04-16 at 3 43 13 PM | Screen Shot 2020-04-16 at 3 43 10 PM |

@Automattic/jetpack-design did we want the explicit branding here on the activity nudge?

On a wpcom site it looks correct on the branch as well (no Jetpack branding). 
Screen Shot 2020-04-16 at 3 43 35 PM

**Comment by keoshi** [comment]
> @Automattic/jetpack-design did we want the explicit branding here on the activity nudge?

I'd say yes for Jetpack sites, no branding for all others if possible. More than the branding though, the border seems more important to fix since the yellow one might convey some sort of warning.

**Comment by sixhours** [comment]
@keoshi I'd started addressing the color/branding issues in #40988; it looks like the `Banner` component was using two different checks for whether to show the jetpack color and jetpack icon, so sometimes they'd be different and we'd show the colors without the icon or vice versa.

Is it OK if we push this through as-is, and I'll continue to address the inconsistencies in the other PR?

**Comment by keoshi** [comment]
@sixhours yeah, absolutely! Thank you.

**Comment by gwwar** [comment]
@sixhours sounds like we can :shipit: and follow up!

**Pull request merged**